dll/win32/shell32/wine/shellord.c

```diff
@@ -1822,20 +1823,24 @@ BOOL WINAPI ReadCabinetState(CABINETSTATE *cs, int length)
 	if ( (r != ERROR_SUCCESS) || (cs->cLength < sizeof(*cs)) ||
 		(cs->cLength != length) )
 	{
+		SHELLSTATE shellstate;
+		shellstate.fWin95Classic = FALSE;
+		SHGetSetSettings(&shellstate, SSF_WIN95CLASSIC, FALSE);
+
 		TRACE("Initializing shell cabinet settings\n");
-		memset(cs, 0, sizeof(*cs));
```

Comment:
why removing it?

Reply:
Because there was only one member that was not being set. We are setting the defaults here and should be specific about everything.

Reply:
This struct has a version member so undefined padding is not a big deal **and** this struct has not changed since NT4 **and** it will take another 8 new bools to end up with hitting the padding...but I will zero it out for you...